streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -269,6 +270,39 @@ class StreamingContext private[streaming] (
     RDDOperationScope.withScope(sc, name, allowNesting = false, ignoreParent = false)(body)
   }
 
+  private[streaming] val recoverableAccuNameToId: mutable.Map[String, Long] = {
```

Comment:
I don't think the memory space is the problem, as in Java collection always stores object references, not the real duplicated objects.

The reason I suggested to do that is to simplify the code at https://github.com/apache/spark/pull/11249/files#diff-f0064bc8820551c338276e29d922e459R47 , which mean we don't need to get the Accumulator again by accumulator ids, particularly the `Accumulators.originals` stores the `WeakReference` object, it can return null.